src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/DropColumns.cs

```diff
@@ -75,23 +78,33 @@ public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType
                 // Verify we have a string literal for the column name. Accd to spec, we don't support
                 // arbitrary expressions that evaluate to string values, because these values contribute to
                 // type analysis, so they need to be known upfront (before DropColumns executes).
-                StrLitNode nameNode;
-                if (nameArgType.Kind != DKind.String || (nameNode = nameArg.AsStrLit()) == null)
+                StrLitNode strLitNode = null;
+                FirstNameNode identifierNode = null;
```

Comment:
>FirstNameNode [](http://example.com/codeflow?start=16&length=13)


Consider an approach like this:

                string value;
                if (!supportIdentifiers)
                {
                    StrLitNode nameNode;
                    if (nameArgType.Kind != DKind.String || (nameNode = nameArg.AsStrLit()) == null)
                    {
                        fArgsValid = false;
                        errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedStringLiteralArg_Name, nameArg.ToString());
                        continue;
                    }

                    value = nameNode.Value;
                } 
                else
                {
                    FirstNameNode identifierNode = nameArg.AsFirstName();
                    if (identifierNode == null)
                    {
                        fArgsValid = false;
                        errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, ????, nameArg.ToString());
                        continue;
                    }

                    value = identifierNode.Ident.Name;
                }

Reply:
- only checks supportIndentifiers once
- simplify the logic. (
- limits the scope of the locals. 
- the current error ErrExpectedStringLiteralArg_Name is probably wrong for identifiers.